## 구구단 출력하기
> 2단부터 9단까지 반복문을 사용하여 구구단을 출력하세요.
---
**[출력 예시]**
```
------- [2 단] -------
2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18

------- [3 단] -------
...
```

In [ ]:
# 아래에 코드를 작성하세요.

In [10]:

for key in range(2, 10):
    print(f'-----[{key}단]-------')
    for dan in range(1, 10):
        print(f'{key} X {dan} = {dan * key}')
    

2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18
-----[2단]-------
3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27
-----[3단]-------
4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36
-----[4단]-------
5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45
-----[5단]-------
6 X 1 = 6
6 X 2 = 12
6 X 3 = 18
6 X 4 = 24
6 X 5 = 30
6 X 6 = 36
6 X 7 = 42
6 X 8 = 48
6 X 9 = 54
-----[6단]-------
7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63
-----[7단]-------
8 X 1 = 8
8 X 2 = 16
8 X 3 = 24
8 X 4 = 32
8 X 5 = 40
8 X 6 = 48
8 X 7 = 56
8 X 8 = 64
8 X 9 = 72
-----[8단]-------
9 X 1 = 9
9 X 2 = 18
9 X 3 = 27
9 X 4 = 36
9 X 5 = 45
9 X 6 = 54
9 X 7 = 63
9 X 8 = 72
9 X 9 = 81
-----[9단]-------


In [12]:
len_phone = ''
while not len_phone:  # 폰번호에 입력값이 있다면 True 입력없다면 False / False 아닐때까지 반복해 / True 일때까지 반복해
    phone_num = input('핸드폰번호를 입력하세요 ')
    len_phone = len(phone_num)  # 폰번호 자리수
    check_010 = phone_num[0:3]  # 폰번호 앞 세자리 
    if len_phone == False:  # 만약 폰번호가 입력되지 않았다면
        continue  # 반복문 다시 돌리세요(폰번호 input 받으세요)
    elif check_010 != '010' or len_phone != 11:  # 만약 폰번호 앞자리가 010이 아니거나 11자리수가 아니라면
        print('휴대폰번호가 아닙니다. 안녕')
    else:
        print('*'*7+ phone_num[7:])  # 어느 경우도 아니라면 앞 7자리는 *로 처리하세요

핸드폰번호를 입력하세요 01012341234
*******1234


## 개인정보보호
> 사용자의 핸드폰번호를 입력 받고, 개인정보 보호를 위하여 뒷자리 4자리를 제외하고는 마스킹 처리하세요.
> * 핸드폰번호는 010으로 시작해야하고 11자리여야한다.
> * 핸드폰번호를 입력하지 않았다면 "핸드폰번호를 입력하세요"를 출력한다.

---

**[입력 예시]**

01012341234

**[출력 예시]**

*******1234

In [ ]:
# 아래에 코드를 작성하세요.

In [4]:
while True:
    phone_num = input()
    if phone_num[0:3] != '010' and len(phone_num) != 11:
        print('핸드폰번호를 다시 입력해주세요')
    else:
        print(phone_num.replace(phone_num[0:11-4],'*'*7))
        break


핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요

핸드폰번호를 다시 입력해주세요
324234234
핸드폰번호를 다시 입력해주세요
01012341234
*******1234


In [1]:
phone_num = input()
if phone_num[0:3] != '010' and len(phone_num) != 11:
    print('핸드폰번호를 다시 입력해주세요')
else:
    print(phone_num.replace(phone_num[0:11-4],'*'*7))

0101231
*******


## 정중앙
> 사용자가 입력한 문자열중 가운데 글자를 출력하세요. 
> * 단, 문자열이 짝수라면 가운데 두글자를 출력하세요.

---

**[입력 예시]**

abc

**[출력 예시]**

b

In [1]:
# 아래에 코드를 작성하세요.

In [4]:
char_str = input()
div = len(char_str)//2

if len(char_str) % 2 == 0:
    print(char_str[div-1:div+1])
else:
    print(char_str[div])

abcdef
cd


## 소수 찾기

> 조건, 반복문을 응용하여 numbers 리스트의 요소들이 소수인지 아닌지 판단하는 코드를 작성하세요.

---


**[입력 예시]**

26, 39, 51, 53, 57, 79, 85

**[출력 예시]**
```python
26 는 소수가 아닙니다. 2 는 26 의 인수입니다.
39 는 소수가 아닙니다. 3 는 39 의 인수입니다.
51 는 소수가 아닙니다. 3 는 51 의 인수입니다.
53 는 소수입니다.
57 는 소수가 아닙니다. 3 는 57 의 인수입니다.
79 는 소수입니다.
85 는 소수가 아닙니다. 5 는 85 의 인수입니다.
```

In [2]:
numbers = [26, 39, 51, 53, 57, 79, 85]
# 아래에 코드를 작성하세요.

In [3]:
demis = [2, 3, 5, 7]
result = 0

for number in numbers:
    for demi in demis:
        if number % demi == 0:
            result = demi
            break
    if number % demi:
        print(f'{number}는 소수입니다.')
    else:
        print(f'{number}는 소수가 아닙니다. {result}는 {number}의 인수 입니다.')

26는 소수가 아닙니다. 2는 26의 인수 입니다.
39는 소수가 아닙니다. 3는 39의 인수 입니다.
51는 소수가 아닙니다. 3는 51의 인수 입니다.
53는 소수입니다.
57는 소수가 아닙니다. 3는 57의 인수 입니다.
79는 소수입니다.
85는 소수가 아닙니다. 5는 85의 인수 입니다.


In [ ]:
#최유진 소수
numbers = [26, 39, 51, 53, 57, 79, 85, 121]
for num in numbers:
    for d_num in range(2, (num // 2)):  # 2부터 확인할 수를 반으로 나눈 값의 -1 까지의 범위
        if num % d_num == 0:  # 26 / 2
            print(f'{num} 는 소수가 아닙니다. {d_num} 는 {num} 의 인수입니다.')
            break
    else:
        print(f'{num} 는 소수입니다.')

In [ ]:
# 최민정
# 소수란, 1과 자신만으로 나누어 떨어지는 1보다 큰 양의 정수

numbers = [26, 39, 51, 53, 57, 79, 85]

primes = [2, 3, 5, 7]  # 소수인지 판별할 숫자 리스트 생성

for number in numbers:
    for prime in primes:
        prime_tf = bool(number % prime)  # 나머지가 있으면 True, 없으면 False로 변환해서 boolean 형태로 저장
        if prime_tf == False:  # 만약 소수가 아니라면,
            print(f'{number} 는 소수가 아닙니다. {prime} 는 {number} 의 인수입니다.')  # 소수가 아니라고 출력
            break
    if prime_tf == True:  # 만약 소수라면,
            print(f'{number} 는 소수입니다.')  # 소수라고 출력

In [16]:
my_numbers = [1, 2, 3, 4, 5, 6]
jackpot_numbers = [2, 1, 4, 5, 6, 7]
bonus_number = 8
cnt = 0
result = ''

for number in my_numbers:
    if number in jackpot_numbers:
        cnt += 1
        
        
if cnt == 5 and bonus_number in my_numbers:
    result = '2등'
elif cnt == 6:
    result = '1등'
elif cnt == 5:
    result = '3등'
elif cnt == 4:
    result = '4등'
elif cnt == 3:
    result = '5등'
        
print(result)

3등
